src/sqlfluff/core/config.py

```diff
@@ -615,7 +615,11 @@ def get(
         self, val: str, section: Union[str, Iterable[str]] = "core", default: Any = None
     ):
         """Get a particular value from the config."""
-        return self._configs[section].get(val, default)
+        section_dict = self.get_section(section)
+        if section_dict is None:
+            return default
```

Comment:
returns default if the section doesn't exist. Could alternatively raise a KeyError if preferred. Fine either way